File: frontend/src/types/EventDTO.ts
import type { TZDate } from "@date-fns/tz";

// basic dto with primative types
export type EventDTO = {
  uuid: string;
  name: string;
  start: string;
  end: string;
  iana_timezone: string;
  colour: string;
  blocking: boolean;
};

// dto with string dates processed into Date objects
export type ProcessedEventDTO = Omit<EventDTO, 'start' | 'end'> & {
  start: TZDate;
  end: TZDate;
};

File: backend/src/schemas/dtos/event_dto.py
from datetime import datetime
from typing import Optional
from dataclasses import dataclass
from uuid import UUID

@dataclass
class EventDTO():
  """  
  EventDTO  
  """
  id: int
  uuid: UUID
  start: datetime
  end: datetime
  iana_timezone: str
  name: str
  colour: str
  created_at: Optional[datetime] = None
  updated_at: Optional[datetime] = None